tests/scripts/run-xunit-perf.py

```diff
@@ -0,0 +1,464 @@
+            'OptLevel',
+            optLevel,
+            '--config',
+            'JitName',
+            jitName,
+            '--config',
+            'PGO',
+            pgoOptimized,
+            '--architecture',
+            architecture,
+            '--machinepool',
+            'PerfSnake']
+    log('')
+    log(" ".join(runArgs))
+
+    proc = subprocess.Popen(runArgs, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    (out, err) = proc.communicate()
+
+    if not proc.returncode == 0:
+        print_error(out, err, 'Createing BenchView submission data failed.')
+
+    if uploadToBenchview:
+        runArgs = [python,
+                os.path.join(benchviewPath, 'upload.py'),
+                'submission.json',
+                '--container',
+                'coreclr']
+        log(" ".join(runArgs))
+        proc = subprocess.Popen(runArgs, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        (out, err) = proc.communicate()
+
+        if not proc.returncode == 0:
+            print_error(out, err, 'Uploading to BenchView failed.')
+
+def verify_core_overlay(coreclrRepo, operatingSystem, arch, configuration):
+    configurationDir = "%s.%s.%s" % (operatingSystem, arch, configuration)
+    overlayDir = 'Core_Root'
+    coreclrOverlay = os.path.join(coreclrRepo, "bin", "tests", configurationDir, "Tests", overlayDir)
+    if not os.path.isdir(coreclrOverlay):
+        raise ValueError("Can't find test overlay directory '%s'. Please build and run CoreCLR tests" % coreclrOverlay)
+
+    return coreclrOverlay
+
+def setup_sandbox(sandboxDir):
+    if os.path.isdir(sandboxDir):
+        shutil.rmtree(sandboxDir)
+    os.mkdir(sandboxDir)
+
+def set_perf_run_log(sandboxOutputDir):
+    if not os.path.isdir(sandboxOutputDir):
+        os.mkdir(sandboxOutputDir)
+    return os.path.join(sandboxOutputDir, "perfrun.log")
+
+def setup_optimization_level(env, optLevel):
+    myEnv = env
+    if optLevel == 'min_opts':
+        myEnv['COMPlus_JITMinOpts'] = '1'
+    elif optLevel == 'tiered':
+        myEnv['COMPLUS_EXPERIMENTAL_TieredCompilation'] = '1'
+
+    return myEnv
+
+def build_perfharness(coreclrRepo, sandboxDir, extension, dotnetEnv):
+    # Confirm dotnet works
+    dotnet = os.path.join(coreclrRepo, 'Tools', 'dotnetcli', 'dotnet' + extension)
+    runArgs = [dotnet,
+            '--info'
+            ]
+    log(" ".join(runArgs))
+
+    proc = subprocess.Popen(runArgs, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=dotnetEnv)
+    (out, err) = proc.communicate()
+
+    if not proc.returncode == 0:
+        print_error(out, err, 'Failed to get information about the CLI tool.')
+
+    # Restore PerfHarness
+    perfHarnessPath = os.path.join(coreclrRepo, 'tests', 'src', 'Common', 'PerfHarness', 'PerfHarness.csproj')
+    runArgs = [dotnet,
+            'restore',
+            perfHarnessPath]
+    log(" ".join(runArgs))
+
+    proc = subprocess.Popen(runArgs, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=dotnetEnv)
```

Comment:
Common case subprocess call. You could also fold in the arg logging logic above to a helper function